### DISCUSSION TITLE
Sitios para aprender más de CSS

### DISCUSSION THREAD

**Discussion by micheldavrdz** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#291] [General]
Como estamos compañeros, ya que estamos viendo CSS esta semana quería compartirles algunas páginas donde pueden aprender un poco más de cómo funciona todo este asunto de las hojas de estilo (y programación en general).

**1.  Can I Use?**

Imagina que estás trabajando en el CSS de un proyecto, investigaste como crear efectos y te encanto como se ve, se lo muestras a tu cliente y **¡oh sorpresa!** algunas cosas no se ven como en tu computadora y descubres que todo tu trabajo lo revisaste en _Google Chrome_ y tu cliente utiliza _Safari_, bienvenid@ al mundo de la compatibilidad entre navegadores.

[**Can I use?**](https://caniuse.com/) nos permite escribir una propiedad de CSS y descubrir en que navegadores funciona, desde que versión y que porcentaje de las personas usando ese navegador tienen esa versión. Si bien las propiedades más utilizadas usualmente sirven en todos los navegadores (dejando a un lado a IE) nunca está de más estar seguros, ya que esto puede pasar con propiedades muy nuevas que solo tienen soporte en Google Chrome (y posiblemente en todos los navegadores basados en Chromium).

También es posible utilizar extensiones para VSCode como [**Autoprefixer**](https://marketplace.visualstudio.com/items?itemName=mrmlnc.vscode-autoprefixer) para hacer nuestra vida un poco más sencilla al añadir automaticamente a nuestro CSS los vendor prefixes necesarios (sobre los cuales puedes aprender más [**aquí**](https://developer.mozilla.org/es/docs/Glossary/Vendor_Prefix)).

**2. MDN Web Docs (Mozilla Developer Network)**

Uno de los sitios más útiles para todo programador web es la documentación de [**MDN Web Docs (Mozilla Developer Network)**](https://developer.mozilla.org/es/), una verdadera joya creada por Mozilla (los creadores de Firefox) y mantenida por la comunidad global donde podemos encontrar información detallada sobre cualquier etiqueta de HTML, propiedad de CSS, sintaxis de JS, etc.

**3. StackOverflow**

La página web por excelencia para cualquier duda de programación. Si alguna vez has tenido un problema en programación y buscas como arreglarlo 9 de 10 veces alguien en [**StackOverflow**](https://stackoverflow.com/) ha tenido ese mismo problema y la comunidad le ha ayudado a resolverlo, por esta misma razon existe la broma de que el día que desaparezca StackOverflow es el día en que se para el mundo jajaja. **Ojo** esto no significa que puedas simplemente copiar y pegar el código y vaya a funcionar, siempre hay que entender que estamos haciendo para saber si la respuesta nos sirven o no. Las preguntas y respuestas están usualmente en inglés, así que deberás dedicar tiempo a aprender el idioma si no te sientes cómod@ leyéndolo (Aunque esto igual deberías hacerlo, ya que la mejor información para programadores casi siempre es en este idioma).

**4. CSS Tricks**

[**Esta página**](https://css-tricks.com) poco a poco se ha vuelto una de mis favoritas al momento de aprender CSS, esta dedicada 100% a enseñar CSS desde conceptos generales hasta como crear efectos específicos. La página está en inglés, así que cae en el mismo caso que StackOverflow.

**5. Flexbox y CSS Grid**

Estos 2 tal vez sean los más atractivos pensando en la actividad de esta semana. Viendo la forma en que está organizada la página a replicar algunos se habrán preguntado "¿Como se supone que acomode todo de esa manera?", y aquí Flexbox y CSS Grid pueden ser lo que están buscando.

Ambos son modelos de layout de CSS3 y permiten crear elementos responsivos dentro de un contenedor que se acomodan automáticamente dependiendo del tamaño de la pantalla (aunque claro siempre es posible necesitar el uso de media queries).

Cada uno tiene sus propias reglas de como utilizarlo, pero esto no significa que sean incompatibles, puedes utilizar ambos en el mismo sitio (diferentes herramientas para diferentes trabajos, ¿verdad?) y suele ser más fácil crear algunos tipos de layout con uno que con el otro.

Si quieren leer y aprender más sobre Flexbox y CSS Grid pueden hacerlo directamente en MDN Web Docs, [**aquí**](https://developer.mozilla.org/es/docs/Web/CSS/CSS_Flexible_Box_Layout/Basic_Concepts_of_Flexbox) para Flexbox y [**aquí**](https://developer.mozilla.org/es/docs/Web/CSS/CSS_Grid_Layout/Basic_Concepts_of_Grid_Layout) para CSS Grid.

Incluso con toda esta información puede ser difícil comprender los conceptos de como acomodar elementos sin verlos, por lo cual existen 2 sitios web donde pueden aprender a utilizarlos con pequeños juegos: [**Flexbox Froggy**](https://flexboxfroggy.com/#es) para aprender Flexbox y [**Grid Garden**](https://cssgridgarden.com/#es) para aprender CSS Grid.

------------------------------------------

Espero que estos sitios web les ayuden a futuro y hagan más placentero su viaje aprendiendo a programar.

**Comment by OscarPedroza** [comment]
Muchas gracias. grandes aportes compañero.

**Comment by hackit92** [comment]
Excelente tu post para los que apenas van conociendo el alcance de CSS, es un gran detalle y ojalá todos pasen para aprovechar tan buenas sugerencias.
